src/server/worker/pipeline/transform/worker.go

```diff
@@ -389,7 +389,7 @@ func forEachDatum(ctx context.Context, driver driver.Driver, baseLogger logs.Tag
 			meta.ImageId = userImageID
 			inputs := meta.Inputs
 			logger := baseLogger.WithData(inputs)
-			env := driver.UserCodeEnv(logger.JobID(), task.OutputCommit, inputs, driver.PipelineInfo().GetAuthToken())
+			env := driver.UserCodeEnv(logger.JobID(), task.OutputCommit, inputs, driver.PipelineInfo().GetAuthToken(), driver)
```

Comment:
@acohen4 having some trouble understanding how we issue a token, this is wrong but hoping you could point me in the right direction. jobs are created by the driver so would we need to reach the api and get a token issued?

Reply:
the token should be issued by PPS. When a Job gets created in propagateJobs(), we should call out to the Auth API Server to create another auth token for the pipeline and set it as the job token

Reply:
@acohen4 thanks, okay i create the token now. so last question here, driver.UserCodeEnv here is called at forEachDatum() do we need a token here to?